puppet/lib/puppet/reports/puppetdb.rb

```diff
@@ -38,7 +38,7 @@ def report_to_hash
       end
 
       resources = build_resources_list
-      is_noop = resources.any? { |rs| has_noop_event?(rs) } and resources.none? { |rs| has_failed_event?(rs) }
+      is_noop = resources.any? { |rs| has_noop_event?(rs) } && resources.none? { |rs| has_enforcement_event?(rs) }
```

Comment:
@wkalt When the report contains no resource events then we cannot mark the whole report as noop. It needs to have at least 1 noop resource event.
So, I'm not sure we can evaluate it just in one cycle.